test/validator.spec.js

```diff
@@ -1210,4 +1220,310 @@ describe('Unit test: validator.js', () => {
+        { evt: 'https://example.com/evt/' },
+      ],
+      type: 'EPCISDocument',
+      schemaVersion: '2.0',
+      creationDate: '2005-07-11T11:30:47.0Z',
+      epcisBody: {
+        eventList: [
+          {
+            eventID: 'test:event:id',
+            type: 'ObjectEvent',
+            action: 'OBSERVE',
+            bizStep: cbv.bizSteps.shipping,
+            disposition: cbv.dispositions.in_transit,
+            epcList: [
+              'urn:epc:id:sgtin:0614141.107346.2017',
+              'urn:epc:id:sgtin:0614141.107346.2018',
+            ],
+            eventTime: '2005-04-03T20:33:31.116-06:00',
+            eventTimeZoneOffset: '-06:00',
+            readPoint: {
+              id: 'urn:epc:id:sgln:0614141.07346.1234',
+              'example:extension': 'factoryId',
+            },
+            bizLocation: {
+              id: 'urn:epc:id:sgln:9529999.99999.0',
+              'evt:factoryId': '8934897894',
+            },
+            bizTransactionList: [
+              {
+                type: cbv.businessTransactionTypes.po,
+                bizTransaction: 'http://transaction.acme.com/po/12345678',
+              },
+            ],
+            sensorElementList: [
+              {
+                'example:furtherEventData': [
+                  { 'example:data1': '123.5' },
+                  { 'example:data2': '0.987' },
+                ],
+                sensorMetadata: { time: '2019-04-02T14:55:00.000+01:00' },
+                sensorReport: [
+                  {
+                    type: cbv.sensorMeasurementTypes.temperature,
+                    value: 26.0,
+                    uom: 'CEL',
+                    deviceID: 'urn:epc:id:giai:4000001.111',
+                    deviceMetadata: 'https://id.gs1.org/giai/4000001111',
+                    rawData: 'https://example.org/giai/401234599999',
+                  },
+                  {
+                    type: cbv.sensorMeasurementTypes.relative_humidity,
+                    value: 12.1,
+                    uom: 'A93',
+                    deviceID: 'urn:epc:id:giai:4000001.222',
+                    deviceMetadata: 'https://id.gs1.org/giai/4000001222',
+                    rawData: 'https://example.org/giai/401234599999',
+                  },
+                ],
+              },
+            ],
+            'example:furtherEventData': [
+              { 'example:data1': '123.5' },
+              { 'example:data2': '0.987' },
+            ],
+          },
+        ],
+      },
+    };
+    let res = {};
+    assert.throws(() => { validateEpcisDocument(epcisDocument); });
+    assert.doesNotThrow(() => { res = validateEpcisDocument(epcisDocument, false); });
+    expect(res.success).to.be.equal(false);
+    expect(res.errors).to.deep.equal(['Event contains unknown extension: example']);
+  });
+  it('should accept a document with an event containing extensions that are not defined in the context (e.g. notInContext:*}'
```

Comment:
what's the difference between this one and the one above?

Reply:
this is for testing all the possibilities of the new features.
In this one you should reject **a document with an event containing extensions** that are not defined in the context
and in the one above you should reject **a document with extensions** that are not defined in the context